packages/veraswap-app/src/atoms/token-balance.ts

```diff
@@ -305,3 +306,120 @@ export const amountOutAtom = atom((get) => {
 
     return amountOut;
 });
+
+// Gets token price in USD
+export const tokenDollarValueAtomFamily = atomFamily(
+    ({ currency, chainId }: { currency: Currency; chainId: number }) =>
+        atomWithQuery<bigint>(() => {
+            if (!currency) return disabledQueryOptions as any;
+
+            return {
+                ...getTokenDollarValueQueryOptions(config, {
+                    tokenAddress: getUniswapV4Address(currency),
+                    chainId,
+                }),
+                // override app default, causes too much flickering
+                staleTime: 5 * 60 * 1000,
+                refetchInterval: 5 * 60 * 1000,
+                refetchOnWindowFocus: false,
+            };
+        }),
+    (a, b) => a.currency.equals(b.currency) && a.chainId === b.chainId,
+) as unknown as AtomFamily<{ currency: Currency; chainId: number }, Atom<AtomWithQueryResult<bigint>>>;
+
+// https://jotai.org/docs/utilities/family#caveat-memory-leaks
+tokenDollarValueAtomFamily.setShouldRemove((createdAt) => Date.now() - createdAt > 5 * 60 * 1000);
+
+export const currenciesBySymbolAtom = atom((get) => {
+    const allCurrencies = get(currenciesAtom);
+    return groupBy(
+        allCurrencies.filter((currency) => currency.symbol),
+        "symbol",
+    );
+});
+
+// Gets the best available token price in USD across all chains
+export const bestTokenDollarValueAtomFamily = atomFamily(
+    (symbol: string) =>
+        atom((get) => {
+            const currencyGroups = get(currenciesBySymbolAtom);
+            const matchingCurrencies = currencyGroups[symbol];
+
+            const quotes = matchingCurrencies.map((currency) => {
+                const quote = get(tokenDollarValueAtomFamily({ currency, chainId: currency.chainId }));
+                return {
+                    currency,
+                    quote: quote.data,
+                    isLoading: quote.isLoading,
+                    isError: quote.isError,
+                };
+            });
+
+            const validQuotes = quotes.filter((q) => q.quote !== undefined && q.quote !== 0n && !q.isError);
```

Comment:
`!!q && !q.isError`

(`!!` handles both undefined and 0n, since both are falsy)